### PR TITLE
ARROW-5430: [Python] Raise ArrowInvalid for pyints larger than int64

### PR DESCRIPTION
**Pull request by kaaveland**
Previously, this case would cause ArrowException: Unknown error, but ArrowInvalid makes more sense and is more likely to be dealt with by callers of pyarrow.array.

### CODE REVIEW

**Review by pitrou — CHANGES_REQUESTED** [review]
Thanks for doing this. Rather than passing `Invalid` explicitly here, I think it would make more sense to recognize `OverflowError` and turn it into `Invalid` automatically. The logic is in the `ConvertPyError` function.

**Comment by kaaveland** [comment]
@pitrou -- changed as suggested.

**Review by pitrou — APPROVED** [review]

**Comment by pitrou** [comment]
Thank you! I'm going to wait for CI before merging.

**Comment by pitrou** [comment]
The MinGW CI failure is unrelated.

**Pull request closed without merging**
